builder/penv_setup.py

```
@@ -36,6 +36,7 @@
     sys.exit(1)
 
 github_actions = bool(os.getenv("GITHUB_ACTIONS"))
+has_network = has_internet_connection() or github_actions
```

Ruff (0.15.5)

[error] Undefined name `has_internet_connection`

(F821)